src/main/java/io/cdap/plugin/gcp/bigquery/sqlengine/input/BigQuerySQLEngineUnshardedExportToCloudStorage.java

```diff
@@ -0,0 +1,93 @@
+ * the License.
+ */
+
+package io.cdap.plugin.gcp.bigquery.sqlengine.input;
+
+import com.google.api.services.bigquery.model.Job;
+import com.google.api.services.bigquery.model.JobConfiguration;
+import com.google.api.services.bigquery.model.JobConfigurationExtract;
+import com.google.api.services.bigquery.model.JobReference;
+import com.google.api.services.bigquery.model.Table;
+import com.google.cloud.hadoop.io.bigquery.BigQueryHelper;
+import com.google.cloud.hadoop.io.bigquery.BigQueryStrings;
+import com.google.cloud.hadoop.io.bigquery.ExportFileFormat;
+import com.google.cloud.hadoop.io.bigquery.UnshardedExportToCloudStorage;
+import io.cdap.plugin.gcp.bigquery.sqlengine.util.BigQuerySQLEngineUtils;
+import org.apache.hadoop.conf.Configuration;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+
+/**
+ * Export which overrides {@link UnshardedExportToCloudStorage} in order to add Snappy compression to Avro exports.
+ */
+public class BigQuerySQLEngineUnshardedExportToCloudStorage extends UnshardedExportToCloudStorage {
+
+  private static final Logger LOG = LoggerFactory.getLogger(BigQuerySQLEngineUnshardedExportToCloudStorage.class);
+
+  public BigQuerySQLEngineUnshardedExportToCloudStorage(Configuration configuration,
+                                                        String gcsPath,
+                                                        ExportFileFormat fileFormat,
+                                                        BigQueryHelper bigQueryHelper,
+                                                        String projectId, Table tableToExport) {
+    super(configuration, gcsPath, fileFormat, bigQueryHelper, projectId, tableToExport, null);
+  }
+
+  /**
+   * Copy of {@link UnshardedExportToCloudStorage#beginExport()} with logic added to enable Snappy compression.
+   */
+  @Override
+  public void beginExport() throws IOException {
+    // Create job and configuration.
+    JobConfigurationExtract extractConfig = new JobConfigurationExtract();
+
+    // Set source.
+    extractConfig.setSourceTable(tableToExport.getTableReference());
+
+    // Set destination.
+    extractConfig.setDestinationUris(getExportPaths());
+    extractConfig.set(DESTINATION_FORMAT_KEY, fileFormat.getFormatIdentifier());
+
+    // Set compression for Avro exports
+    if (fileFormat == ExportFileFormat.AVRO) {
+      extractConfig.setCompression(BigQuerySQLEngineUtils.CODEC_SNAPPY);
+    }
+
+    JobConfiguration config = new JobConfiguration();
+    config.setExtract(extractConfig);
+
+    JobReference jobReference =
+      bigQueryHelper.createJobReference(
+        projectId, "exporttocloudstorage", tableToExport.getLocation());
+
+    Job job = new Job();
+    job.setConfiguration(config);
+    job.setJobReference(jobReference);
+
+    // Insert and run job.
+    try {
+      Job response = bigQueryHelper.insertJobOrFetchDuplicate(projectId, job);
+      LOG.debug("Got response '{}'", response);
```

Comment:
what ends up getting logged here? Would be better to be more specific with the message than 'response'.

Reply:
This is a straight copy from the underlying implementation: https://github.com/adam-stasiak/bigdata-interop/blob/master/bigquery/src/main/java/com/google/cloud/hadoop/io/bigquery/AbstractExportToCloudStorage.java#L75

Since Job extends from `GenericJson`, this response is just the JSON payload for this response object.